pallet/ethtx-forwarder/src/lib.rs

```diff
@@ -259,8 +259,35 @@ impl<T: Config> Pallet<T> {
 		Ok(transaction)
 	}
 
+	pub fn forward_call(
+		from: H160,
+		to: H160,
+		data: Vec<u8>,
+		value: U256,
+		gas_limit: U256,
+	) -> Result<pallet_evm::CallInfo, sp_runtime::DispatchError> {
+		let (who, _) = pallet_evm::Pallet::<T>::account_basic(&from);
+		<T as pallet_evm::Config>::Runner::call(
+			from,
+			to,
+			data,
+			value,
+			gas_limit.unique_saturated_into(),
+			None,
+			None,
+			Some(who.nonce),
+			vec![],
+			false, // non-transactional
+			true,
+			None,
+			None,
+			<T as pallet_evm::Config>::config(),
+		)
+		.map_err(|err| err.error.into())
+	}
+
 	#[cfg(feature = "evm-tracing")]
-	fn trace_tx(
+	fn trace_apply(
```

Comment:
Sounds strange; "trace on" or "trace at" sounds better.

Reply:
The primary rationale behind this is `T::ValidatedTransaction::apply`.  I aim to indicate that this is a tracing version of the apply function.

Reply:
Make sense. I keep the original naming.